src/EnergyPlus/main.cc

```diff
@@ -50,21 +50,24 @@
 #include <EnergyPlus/Data/EnergyPlusData.hh>
 #include <EnergyPlus/DataStringGlobals.hh>
 #include <EnergyPlus/api/EnergyPlusPgm.hh>
+#include <EnergyPlus/api/state.h>
 
 int main(int argc, const char *argv[])
 {
-    EnergyPlus::EnergyPlusData state;
+    EnergyPlus::EnergyPlusData *state = reinterpret_cast<EnergyPlus::EnergyPlusData*>(stateNew());
     // these need to be set early to be used in help and version output messaging
     // this was pulled from EnergyPlusPgm.cc and needs to be removed once the release is done
     Array1D_int value(8);
     std::string datestring; // supposedly returns blank when no date available.
     date_and_time(datestring, _, _, value);
     if (!datestring.empty()) {
-        state.dataStrGlobals->CurrentDateTime = fmt::format(" YMD={:4}.{:02}.{:02} {:02}:{:02}", value(1), value(2), value(3), value(5), value(6));
+        state->dataStrGlobals->CurrentDateTime = fmt::format(" YMD={:4}.{:02}.{:02} {:02}:{:02}", value(1), value(2), value(3), value(5), value(6));
     } else {
-        state.dataStrGlobals->CurrentDateTime = " unknown date/time";
+        state->dataStrGlobals->CurrentDateTime = " unknown date/time";
     }
-    state.dataStrGlobals->VerStringVar = EnergyPlus::DataStringGlobals::VerString + "," + state.dataStrGlobals->CurrentDateTime;
-    EnergyPlus::CommandLineInterface::ProcessArgs(state, argc, argv);
-    return EnergyPlusPgm(state);
+    state->dataStrGlobals->VerStringVar = EnergyPlus::DataStringGlobals::VerString + "," + state->dataStrGlobals->CurrentDateTime;
+    EnergyPlus::CommandLineInterface::ProcessArgs(*state, argc, argv);
+    auto ep_ret = EnergyPlusPgm(*state);
+    stateDelete(reinterpret_cast<EnergyPlusState>(state));
```

Comment:
So is this _actually_ necessary for `main()` to go through the process of cleaning up state when the next line will terminate the program?

Reply:
An intentional memory leak is _actually_ still a memory leak. I don't love the stateDelete and casting shenanigans, but I'd prefer at least a comment if we remove them and rely on termination cleanup. 
These functions happened to be available to us, but are clearly intended for binding through a C interface. The whole thing could be a lot less awkward if the state manipulation in main() could be moved into the DLL entirely. Or, if we had an RAII EnergyPlusState generator class exported from the DLL.